bindings/native/go/lib.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+package stronghold
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/awnumar/memguard"
+	"golang.org/x/crypto/blake2b"
+)
+
+type StrongholdNative struct {
+	ptr     StrongholdPointer
+	enclave *memguard.Enclave
+}
+
+func NewStronghold(key string) *StrongholdNative {
+	stronghold := &StrongholdNative{}
+	stronghold.enclave = memguard.NewEnclave([]byte(key))
+	return stronghold
+}
+
+// SetLogLevel
+/**
+  0 => LevelFilter::Off
+  1 => LevelFilter::Error
+  2 => LevelFilter::Warn
+  3 => LevelFilter::Info
+  4 => LevelFilter::Debug
+  5 => LevelFilter::Trace
+*/
+
+func SetLogLevel(level int) {
+	setLogLevel(level)
+}
+
+func (s *StrongholdNative) validate() error {
+	if s.ptr == nil {
+		return errors.New("snapshot is unavailable. Call Open/Create")
```

Comment:
The error message is too specific and bound to `Snapshots`. Would it be possible to pass in an error object, in case it fails?